kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyExceptionMapper.java

```diff
@@ -196,16 +196,26 @@ public static ApiMessage errorResponseMessage(DecodedRequestFrame<?> frame, Thro
         return errorResponse(frame, error).data();
     }
 
+    public static AbstractResponse errorResponseForMessage(ApiMessage message, Throwable error) {
+        final short apiKey = message.apiKey();
+        return errorResponse(ApiKeys.forId(apiKey), message, message.highestSupportedVersion()).getErrorResponse(error);
```

Comment:
not so sure about pulling the highest supported version for the MessageType here. This looks like it will return the highest version the proxy can handle, for example on `ProduceRequestData`:

```
@Override
    public short highestSupportedVersion() {
        return 11;
    }
    ```
    
I think we need the version the client sent on the request, think it'll be in the RequestHeaderData

Reply:
Resolved with d4201bf1bb6e52e26204e81a599a75de9b53e5a0